core/migration/migrate_external_integration.py

```diff
@@ -0,0 +1,137 @@
+import json
+from collections import defaultdict
+from typing import Dict, Tuple, Type, TypeVar
+
+from sqlalchemy.engine import Connection, CursorResult, Row
+
+from core.integration.settings import (
+    BaseSettings,
+    ConfigurationFormItemType,
+    FormFieldInfo,
+)
+from core.model import json_serializer
+
+T = TypeVar("T", bound=BaseSettings)
+
+
+def _validate_and_load_settings(
```

Comment:
It's not an exact copy, I've broken up the code to be more modular so it can be used for the other kinds of integration migrations. 
The Auth migration did some special things within these methods specially for Auth which I've removed.